src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1680,7 +1862,32 @@ public override BoundNode VisitSwitchDispatch(BoundSwitchDispatch node)
                     casesBuilder.Add((value, GetLabelClone(label)));
                 }
 
-                return node.Update(expression, casesBuilder.ToImmutableAndFree(), defaultClone);
+                var lengthBasedSwitchData = node.LengthBasedStringSwitchDataOpt;
+                if (lengthBasedSwitchData is not null)
+                {
+                    Debug.Assert(false, "This is unreachable at the moment");
```

Comment:
>Debug.Assert(false, "This is unreachable at the moment"); [](http://example.com/codeflow?start=20&length=57)

When is `FinallyCloner` used (and why isn't this code reachable)?

Should report an emit error here if we reach this code path rather than relying on untested code?

Reply:
It's only used for a `fault` block (`PreferFaultHandler` set). This flag is only set by `SyntheticBoundNodeFactor.Fault(...)` which is only used by `IteratorMethodToStateMachineRewriter` for the disposal (see `GenerateMoveNextAndDispose`, line 146). The block we generate for this is only `this.Dispose()`.
So most of the code in `FinallyCloner` is not exercised :-/
Given that, I chose to also implement this case, but put an assertion to flag any test that ends up hitting it some day... We can discuss Monday.